File: programmers/java/lev0/특정 문자열로 끝나는 가장 긴 부분 문자열 찾기.java
public class 특정 문자열로 끝나는 가장 긴 부분 문자열 찾기 {
    
}
class Solution {
    public String solution(String myString, String pat) {
        int idx = myString.lastIndexOf(pat);
        return myString.substring(0, idx+pat.length());
    }
}

File: programmers/js/lev0/특정 문자열로 끝나는 가장 긴 부분 문자열 찾기.js
function solution(myString, pat) {
  let idx = myString.length - pat.length;
  while (idx >= 0) {
    if (myString.substring(idx, idx + pat.length) === pat) break;
    idx -= 1;
  }
  return myString.substring(0, idx + pat.length);

  //     return str.substring(0, str.lastIndexOf(pat)) + pat;\
}
